subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c

```diff
@@ -1342,26 +1331,55 @@ gst_msdkenc_set_src_caps (GstMsdkEnc * thiz)
   return TRUE;
 }
 
+static GstBufferPool *
+gst_msdk_create_va_pool (GstMsdkEnc * thiz, guint num_buffers)
+{
+  GstBufferPool *pool = NULL;
+  GstAllocator *allocator;
+  GArray *formats;
+  GstAllocationParams alloc_params = { 0, 31, 0, 0 };
+  GstVaDisplay *display = NULL;
+  GstCaps *aligned_caps = NULL;
+  GstVideoInfo info = thiz->input_state->info;
+
+  display = (GstVaDisplay *) gst_msdk_context_get_display (thiz->context);
+
+  formats = g_array_new (FALSE, FALSE, sizeof (GstVideoFormat));
+  g_array_append_val (formats, GST_VIDEO_INFO_FORMAT (&info));
+
+  /* Currently, we only use va_allocator to create va pool, this is because
+   * with va_dmabuf_alllocator we will encounter modifer issue.
+   */
+  allocator = gst_va_allocator_new (display, formats);
+
+  if (!allocator) {
+    GST_ERROR_OBJECT (thiz, "failed to create allocator");
```

Comment:
formats is not freed

Reply:
Va allocator has a pointer pointing to this format array.  When the allocator is freed, the format pointer will also freed. Please correct me if I made a misunderstanding.

Reply:
format should be freed in the error path because allocator is NULL.

Reply:
> formats should be freed in the error path because allocator is NULL.

Right, I missed the case of the allocator being NULL. Thanks, will fix this.